addons/mrp/static/src/components/bom_overview/mrp_bom_overview.js

```diff
@@ -48,19 +48,29 @@ export class BomOverviewComponent extends Component {
     //---- Data ----
 
     async initBomData() {
-        const bomData = await this.getBomData();
+        const bomData = await this.getBomData(true);
         this.state.bomQuantity = bomData["bom_qty"];
         this.state.showOptions.uom = bomData["is_uom_applied"];
         this.uomName = bomData["bom_uom_name"];
         this.variants = bomData["variants"];
         this.showVariants = bomData["is_variant_applied"];
         if (this.showVariants) {
-            this.state.currentVariantId = Object.keys(this.variants)[0];
+            const variantId = this.props.action.context.active_product_id;
+            if(variantId && this.state.currentVariantId != variantId){
+                this.onChangeVariant(variantId);
+            }
```

Comment:
I'm not sure why you do this. From what I could see, for now `this.state.currentVariantId` is either `this.props.action.context.active_product_id` (if it was defined) or `false`.
How could it change to something different than `this.props.action.context.active_product_id` by then?
Maybe I'm missing something?